### ISSUE TITLE
NS_ERROR_IN_PROGRESS error in netCacheReader.js

### ISSUE THREAD

**Issue by umanamente** [firebug/firebug#8003]
I have tons of NS_ERROR_IN_PROGRESS messages in my console:
```
NS_ERROR_IN_PROGRESS: Component returned failure code: 0x804b000f (NS_ERROR_IN_PROGRESS) [nsICacheEntry.dataSize] 
```
In [netCacheReader.js:307:0](https://github.com/firebug/firebug/blob/3a9367d18f094e2f0a70f4e001e7066f955c8092/extension/content/firebug/net/netCacheReader.js#L307)

I have Firefox 45.0.1 (also occured in previous versions)

**Comment by janodvarko** [comment]
What Firebug version do you use?
Honza

**Comment by umanamente** [comment]
2.0.15 (latest)

**Comment by markusp72** [comment]
+1

**Comment by lserni** [comment]
Same versions as markusp72, same symptoms. They disappear if I either disable the browser cache, or the whole Network tab altogether.

The exact error says,
NS_ERROR_IN_PROGRESS: Component returned failure code: 0x804b000f (NS_ERROR_IN_PROGRESS) [nsICacheEntry.dataSize]

and is reported in netCacheReader.js at line 307 (where it does actually attempt to access descriptor.datasize).

**Comment by lserni** [comment]
I have manually modified the Firebug XPI file and patched netCacheReader.js:

+++    var dds = descriptor.dataSize || 0;

    file.cacheEntry =
    [
        {
            name: "Last Modified",
            value: NetUtils.getDateFromSeconds(descriptor.lastModified)
        },
        {
            name: "Last Fetched",
            value: NetUtils.getDateFromSeconds(descriptor.lastFetched)
        },
        {
            name: "Expires",
            value: NetUtils.getDateFromSeconds(descriptor.expirationTime)
        },
        {
            name: "Data Size",
---            value: descriptor.dataSize
+++            value: dds

        },

So far, everything appears to be working. I _did_ have to disable extension signature verification since, of course, the package now shows as tampered.

**Comment by lserni** [comment]
I have wrapped the assignment in netCacheReader.js inside a try block, and **the error has disappeared**.

Of course I had to set extension signature checking to false, or Firefox would have rejected the tampered package (this also means that the error will reappear in Firefox v47 when the signature checking will be made mandatory).

**Comment by janodvarko** [comment]
Can you send PR?

Honza

**Comment by lserni** [comment]
I'm really sorry, Honza. I thought I had signed up for notifications but somehow yours never did reach me. I'm not too conversant with git or github, so I can't quickly whip up a pull request. However the modification I did was in 

    firebug/extension/content/firebug/net/netCacheReader.js

where I changed

    value: descriptor.dataSize

in line 307 with

    value: descriptorDataSize

then, at line 290, I declared descriptorDataSize and assigned it descriptior.dataSize within a try block, setting it to zero in case of failure. Funnily enough, I did not see that zero pop up anywhere - it's just as if the code was executed *twice*, the first time throwing an error, the second time filling in the expected values. But understand that I haven't debugged it at all, this is only my impression.

**Comment by janodvarko** [comment]
Thanks for the update!

Patch committed.

Honza

**Comment by envision** [comment]
Great to have a fix for this. I've seen these for a long time but looked up to this only now...

**Comment by envision** [comment]
@janodvarko I'm still getting a ton of these errors (for every API request made by a front page app) and have Firebug 2.0.16, was this supposed to be fixed already in this version?

    Component returned failure code: 0x804b000f (NS_ERROR_IN_PROGRESS) [nsICacheEntry.dataSize]

**Comment by janodvarko** [comment]
It's fixed in Firebug 2.0.17 that has been just released.

Honza
